megatron/core/extensions/transformer_engine.py

```diff
@@ -706,6 +706,15 @@ def forward(
         else:
             return core_attn_out
 
+    def sharded_state_dict(self, prefix='', sharded_offsets=(), metadata=None):
+        state_dict = self.state_dict(prefix='', keep_vars=True)
```

Comment:
Do we just need `state_dict = self.state_dict(prefix, keep_vars=True)` here? The `prefix` seems already having a default value.

Reply:
Done. Thanks

Reply:
Changed back to using prefix='', in order to follow the same code path as in function sharded_state_dict_default from https://github.com/ROCm/Megatron-LM/blob/d9a6c85a4e475b274cae6b0a0d58b22306c701a5/megatron/core/transformer/utils.py#L184